### PR TITLE
Closes #156 assert character scalar does not give expected message when case sensitive equals false

### LINKED ISSUE #156: Bug: assert_character_scalar() does not give expected message when case_sensitive = FALSE
### What happened?

Hi @thomas-neitmann and @bms63 , I created a workaround for this issue in my local area, and now I cannot reproduce the issue, even when I restart R and clear my global environment. However, I'm sure I didn't imagine it, so here is the issue and my suggested solution:

For `assert_character_scalar()`, when I assigned `case_sensitive = FALSE`, if I then specify a value that does not match the list of valid values, then instead of the intended error message, the code seemed to fall over where arg_name() was called at line 167. I think this is something to do with assigning  `arg <- tolower(arg)`, so the value of arg is modified and substitute(arg) does not return the parse tree. I was able to get my code working as follows:

- I made a copy of assert_character_scalar()
- I commented out lines 160 to 162 (line numbers referring to where this function is located in assertions.R)
- I replaced line 164 with the code below.

```
  if (!is.null(values) && (case_sensitive && arg %notin% values ||
                           !case_sensitive && tolower(arg) %notin% values)) {

```
Here is the code that I was running, in which I previously observed the issue (even though I'm no longer able to reproduce the issue):

```
age_unit <- "month"

  age_unit <- assert_character_scalar(
    age_unit,
    values = c(
      "years", "months", "weeks", "days",
      "hours", "minutes", "seconds"
    ),
    case_sensitive = FALSE,
    optional = TRUE
  )

```

### Session Information

_No response_

### Reproducible Example

_No response_

### PR DESCRIPTION
**Pull request by fshanlee**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/assertions.R [hunk 1]
```diff
@@ -158,14 +158,19 @@ assert_character_scalar <- function(arg,
     abort(err_msg)
   }
 
-  if (!case_sensitive) {
-    arg <- tolower(arg)
+  if (case_sensitive) {
```
Comment (bundfussr) [inline R/assertions.R:161]: Could you add a comment regarding why `case_adjusted_arg` is used?

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Review by bundfussr — APPROVED** [review]

**Comment by fshanlee** [comment]
I've made another amendment to assert_character_scalar() and its corresponding testthat scripts: previously, it appeared that the code was attempting to return the function's input in lowercase if `case_sensitive = FALSE`. However, having looked at other functions in assertions.R, I think it would be more consistent to return the input in its original case, regardless of the value of case_sensitive, and indeed the roxygen header for assert_character_scalar() states "_... Otherwise, the input is returned invisibly._" and does not mention changing the case of what is returned depending on case_sensitive. As an aside, I noticed that there are some functions in assertions.R that don't return their input invisibly, even when the roxygen header states that they do--I've opened issue #167 to address this.

**Comment by bundfussr** [comment]
@fshanlee , please keep the return value as it is, i.e., return lower case if `case_sensitive = FALSE`. We use this to make the argument "case insensitive". If there are conditions on the value in the function code, we use lowercase, e.g.,
```
mode <- assert_character_scalar(mode, case_sensitive = FALSE)
if (mode == "first") {
...
}
```

**Comment by fshanlee** [comment]
> @fshanlee , please keep the return value as it is, i.e., return lower case if `case_sensitive = FALSE`. We use this to make the argument "case insensitive". If there are conditions on the value in the function code, we use lowercase, e.g.,
> 
> ```
> mode <- assert_character_scalar(mode, case_sensitive = FALSE)
> if (mode == "first") {
> ...
> }
> ```

Hi @bundfussr , I reverted the last commit, so the function will always return lower case when `case_sensitive = FALSE`.

**Comment by bundfussr** [comment]
@fshanlee , looks good to me.

**Pull request merged**